First pass edited of draft doc.

hjoliver opened the pull request:
@oliver-sanders - try this for size (just the overview section).  Cover email coming...

hjoliver committed:
Edited overview section.

hjoliver committed:
Intro mods: suite.rc, H World, and Tasks sub-sections.

oliver-sanders commented:
`! LaTeX Error: Environment shaded* undefined.`

hjoliver commented:
Huh, worked for me. Should be defined by the 'framed' package.  I'll check
again when I'm home later...

On 25/08/2016 8:20 PM, "Oliver Sanders" notifications@github.com wrote:

> ! LaTeX Error: Environment shaded\* undefined.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/oliver-sanders/cylc-tutorial/pull/1#issuecomment-242313901,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/ACQLGSSuRLbhvtg-8p9t2iPpq_y7bMukks5qjVBmgaJpZM4JryiF
> .

hjoliver commented:
Yes, `shaded` is defined by `\usepackage{framed}` in the document preamble.   On my metomi ubuntu 14.04 VM  (can't recall if I installed LaTeX myself on it or not), the front page of `/usr/share/doc/texlive-doc/latex/framed/framed.pdf` confirms this (and I don't get the error).  Do you have `framed` installed in your environment?

hjoliver committed:
Disable use of framed package.

oliver-sanders commented:
We have `/usr/share/texmf/tex/latex/ltxmisc/framed.sty` which looks like something different all-together.

hjoliver commented:
Ok, just pushed a commit to disable the framed package.  My new text blocks are still visible due to the grayed out text color.

oliver-sanders commented:
I like the new content, the document is becoming much more informative, however, I'm not sure all this detail is suitable for Lisbon. We had this in mind as more of a first steps in cylc than a comprehensive introduction for new users (though a longer more detailed version would be useful for other purposes).

There is also quite a lot of cylc terminology here (e.g. `task pool`, `daemon`) which I has tried to avoid as we can't be sure what previous experience people have and also it will be an international audience (English won't be the first language).

We could move some of the detail to the advanced tutorials section to keep things light (we were worried that it was getting a bit long in the first place).

oliver-sanders commented on cylc-overview.tex:
```diff
@@ -11,38 +8,74 @@ \section*{Cylc Tutorial}
+submit their jobs to run once their inputs have been satisfied by the
+completed outputs of other tasks.
+
+A {\em task} is an abstract representation of a {\em job} that runs on a
```
What is a job?

hjoliver replied:
The sentence says, "a job that runs on a computer".   Is there any other English term to describe that concept?   And in the context of cylc, the fact that a task is not a job, but it runs one...

benfitzpatrick replied:
Maybe a 'script'?

hjoliver replied:
I used "job" to cover both scripts and binary programs.  (You wouldn't call the UM a "script" :grin: ).  However, arguably "script" does cover it for cylc, since all jobs are wrapped in a job script.

oliver-sanders commented on intro-section.tex:
```diff
@@ -3,152 +3,272 @@ \section{Cylc Introduction}
+        \item
+    \lstinline{cylc-register} just associates a name with a suite definition.
+    Commands that parse a suite can refer to this suite name or the file path;
+    commands that connect to a suite daemon must refer to the suite name.
```
Do we need the second two lines?

hjoliver replied:
This is in a "more info" box, and it's there to explain _why_ registration is needed, and what a "suite name" is used for.    If we run through the tutorial without explaining things like this will the users understand what they're doing?    It seems to me, if I was told "you need to register a suite before you can run it", my next thought would be "what does that mean, and why do it need to do it?".

oliver-sanders commented on intro-section.tex:
```diff
@@ -3,152 +3,272 @@ \section{Cylc Introduction}
+that it has succeeded.  Once the task has succeeded, the suite has no more
+tasks to run and will automatically shut down.
+
+The task job script and its standard out and standard error are captured to
```
This seems a bit heavy for hello_world, could It be moved to a later point or into the advanced tutorials section?

hjoliver replied:
Maybe my terminology is too heavy, but surely if a user runs a Hello World program and does not see "Hello World!" printed at the terminal, they are going to wonder where exactly it was printed.  Which requires explaining where the output goes.  I suppose how it got there could be skipped, or relegated to a "more info" box, or even punted to an appendix.

oliver-sanders replied:
>  they are going to wonder where exactly it was printed

Very true that should be explained in hello world, could we write a quick explanation then move the detail into the advanced tutorial?

oliver-sanders commented on intro-section.tex:
```diff
@@ -3,152 +3,272 @@ \section{Cylc Introduction}
 
-To convert a script into a cylc task the script should be placed inside a
-`bin` directory alongside the suite.rc file in the suite directory.
+Any program or script can be used as a cylc task without modification so long as,
```
Advanced tutorial?

hjoliver replied:
The original sentence was wrong.  (a) you don't need to "convert" your existing scripts in any way in order to automate their use with cylc - important because it speaks to the simplicity (well, relatively speaking!) of the system; and (b) and they don't need to be put in the suite bin directory either - that is just one option.    Perhaps you're referring mainly to the next line, the two conditions for use (error status, and no detached internal processes).  I suppose that could be shifted.

oliver-sanders replied:
Sorry, it was a bit misleading commenting on this line, I was referring to the following content. It just feels a little distracting from the tutorial.

oliver-sanders commented on intro-section.tex:
```diff
@@ -3,152 +3,272 @@ \section{Cylc Introduction}
+            batch system = pbs
+\end{lstlisting}
+
+\more{Cylc can submit jobs to various batch systems such as PBS and SLURM. With
```
Getting a bit advanced.

hjoliver replied:
I would have thought everyone present will want to know how to submit jobs to an HPC, and this shows its pretty easy to do.  Maybe you're right on grounds that there just isn't time to cover everything, however.

benfitzpatrick replied:
I agree with Hilary here - a large part of cylc is abstracting away interacting with batch submission. I think it's worth it, given that we can't actually do an advanced tutorial (that would run) about it.

oliver-sanders commented on intro-section.tex:
```diff
@@ -3,152 +3,272 @@ \section{Cylc Introduction}
+[runtime]
+    [[model]]
+        script = run-model
+        [[[environment]]]
```
Introduce the `[[[environment]]]` section.

oliver-sanders commented on preamble.tex:
```diff
@@ -64,3 +63,14 @@
+%        \begin{shaded*}
+\textbf{More:} #1
+%        \end{shaded*}
+    }}
```
Could this be consolidated with `\note`?

hjoliver replied:
I suppose so, although the intended purpose is quite different.

hjoliver commented:
> I'm not sure all this detail is suitable for Lisbon. We had this in mind as more of a first steps in cylc than a comprehensive introduction for new users (though a longer more detailed version would be useful for other purposes).

Good point, maybe I went too far ... but I guess I was trying to put some of this more comprehensive explanatory material into the new "more info" boxes which don't really have to be read immediately, but are there for the curious; _and_ I don't think we can leave obvious questions unexplained, such as (for example) "where is the output that says Hello World!" when they run the Hello World suite.

hjoliver commented:
> There is also quite a lot of cylc terminology here (e.g. task pool, daemon) which I has tried to avoid as we can't be sure what previous experience people have and also it will be an international audience (English won't be the first language).

Another good point.  We should definitely avoid jargon where possible, or else explain what it means.  What's the alternative to "daemon" for instance?   "Suite server" is ambiguous as it can refer to the host machine.  Why does no output go to the terminal, and does the suite get killed if I log out?  The correct term covers all that, although I agree it might need explaining to some (if it can't be avoided).

Also, I think the tutorial should stand alone, for use by others when we're not there to explain missing details.  

Of course there's no point writing a tutorial that we can't possibly get through in a day though...

oliver-sanders commented:
> I was trying to put some of this more comprehensive explanatory material into the new "more info" boxes which don't really have to be read immediately

This is a good idea, it would be nice if we could move them out of line with the rest of the content (say into a column / margin alongside the main content like `\marginnote`) so that tutorial still reads quickly.

hjoliver commented:
>  it would be nice if we could move them out of line with the rest of the content ...

Yep, good idea, I'll see what I can do this arvo.

hjoliver committed:
Rewrote my changes in light of discussions.

hjoliver commented:
Rewrote my changes to the first two sections (cylc overview + cylc intro):
- added a short "Tutorial Scope" page at the start, which also introduces Rose to explain why we're using two Rose utilities in a cylc tutorial (also for the Rose section later on).
- added some highlighted terminology definition lines
- margins don't provide enough space for important but non-essential extra info, so I added a couple of section-specific appendices
  - this includes the job schedule diagrams of the "cylc overview" - I think the only important thing to convey in the overview is how cylc handles cycling workflows differently from other systems.
- show where to find job output after running the Hello World demo
-  introduced chaining of dependencies and use of '&' in graphs from the start.  IMO these aren't advanced features, they are intuitive because they make graph strings look more like the graph that they are defining.  Otherwise, a mere list of dependency pairs looks quite unsophisticated and verbose. 
- The initial "Cycling Workflow" example:
  - used `[[[P1M]]]` instead of `[[[T00]]]` because the audience will be climate modellers, and new users would wonder how it is that `T00` (which is not an interval) can define a sequence.
  - changed the description to emphasize that the inter-cycle triggers are to reflect real dependence in the tasks, and that if there is no such dependence then cylc can (and should) be able to run the tasks concurrently

The main tutorial content is now no longer than it was originally, not counting the new "Tutorial Scope" page.

benfitzpatrick commented:
I'm happy with this, especially now that we have appendices.

benfitzpatrick commented:
But you need to add your name to the authors list! I'm content to be third author

oliver-sanders commented:
This feels deliverable! I'll add @hjoliver to the authors list outside of this PR.
